swiftsimio/initial_conditions/generate_particles.py

```diff
@@ -1,5 +1,1304 @@
 """
-Particle generation code.
+Generate SPH initial conditions for SPH simulations iteratively for a given density function following Arth et al.
+2019 (https://arxiv.org/abs/1907.11250).
 
-TBD
 """
+import numpy as np
+import unyt
+from math import erf
+from typing import Union
+import warnings
+
+from .IC_kernel import get_kernel_data
+from swiftsimio.optional_packages import KDTree, TREE_AVAILABLE
+from swiftsimio import Writer
+
+
+class RunParams(object):
+    r"""
+    A class to store iteration runtime parameters. Before running the iteration,
+    these parameters are intended to be direcly accessed by the user.
+
+
+    Attributes
```

Comment:
Styling request: please style this so that there are the same number of dashes as characters, and have a space between the items and the attributes so:

```
Attributes
----------

iter_max: int
```

Reply:
Done.